lib/measured/conversion.rb

```diff
@@ -48,6 +48,9 @@ def conversion_table
 
   def unit_for(name)
     unit = @units.find { |unit| unit.names_include?(name, case_sensitive: @case_sensitive) }
-    unit or raise Measured::UnitError, "Cannot find unit for #{name}"
+  end
+
+  def unit_for!(name)
+    unit_for(name) or raise Measured::UnitError, "Cannot find unit for #{name}"
```

Comment:
I'm okay with not using `or`, but a couple of my own thoughts that I wouldn't mind some thoughts on before not using it:

1. The style guide takes a dogmatic approach, banning it without reason. I am vehemently against any style guide statement that lacks reasoning (unless it's obvious, which in this case it isn't). 
2. I've seen style guides that allow it for control flow but not as a boolean operation, which makes a lot more sense. I actually very much disagree with the style guide suggestion of using `||` for control flow, because the precedence of `and` and `or` is very close to that of modifier `if` and modifier `unless`.

tl;dr I don't (completely) agree with the style guide, and in some sense would argue that it's contradicting itself by recommending modifier if/unless but blindly banning or/and.